src/fapolicy-cli.c

```diff
@@ -0,0 +1,99 @@
+/*
+ * fapolicy-cli.c - CLI tool for fapolicyd
+ * Copyright (c) 2016,2018 Red Hat Inc., Durham, North Carolina.
+ * All Rights Reserved.
+ *
+ * This software may be freely redistributed and/or modified under the
+ * terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2, or (at your option) any
+ * later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; see the file COPYING. If not, write to the
+ * Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor
+ * Boston, MA 02110-1335, USA.
+ *
+ * Authors:
+ *   Radovan Sroka <rsroka@redhat.com>
+ */
+
+
+#include <stdio.h>
+#include <string.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <errno.h>
+
+const char * usage =
+"Fapolicyd CLI Tool\n\n"
+"help       Prints this help message\n"
+"update     Notifies fapolicyd to perform update of database\n"
+;
+
+const char * _pipe = "/var/lib/fapolicyd/fapolicyd.fifo";
+
+int main(int argc, char const *argv[])
+{
+    if (argc == 1) {
+        fprintf(stderr, "Too few arguments\n\n");
+        fprintf(stderr, "%s", usage);
+        return 1;
+    }
+
+    if (argc > 2) {
+        fprintf(stderr, "Too many arguments\n\n");
+        fprintf(stderr, "%s", usage);
+        return 1;
+    }
+
+    if (strcmp(argv[1], "help") == 0) {
+        printf("%s", usage);
+        return 0;
+    } else if (strcmp(argv[1], "update") == 0) {
+        int fd = -1;
+        struct stat s;
+
+        if (stat(_pipe, &s) == -1) {
+            fprintf(stderr, "Stat: %s -> %s\n", _pipe, strerror(errno));
+            return 1;
+        } else {
+            if (!S_ISFIFO(s.st_mode)) {
+                fprintf(stderr, "File: %s exists but it is not a pipe!\n", _pipe);
+                return 1;
+            }
+        }
+
+        fd = open(_pipe, O_WRONLY);
```

Comment:
I'd suggest swapping the stat and open operations. If you open it and then fstat the file, then you do not have a TOCTOU race where an attacker might put a symlink to something important.